lib/mayaUsd/commands/Readme.md

```diff
@@ -0,0 +1,441 @@
+| [1, 3]       | 2.0         | -3.0, 3.0      | -2.0, -1.0, 4.0, 5.0                             |
+
+The last example is quite strange, and a warning will be issued.
+This is how it will be processed:
+
+* The current time starts at 1.0. We evaluate frame samples,
+  giving us time samples -2.0 and 4.0.
+* Then we advance the current time by the stride (2.0),
+  making the new current time 3.0. We evaluate frame samples,
+  giving us time samples -1.0 and 5.0.
+* The time samples are sorted before exporting, so they are
+  evaluated in the order -2.0, -1.0, 4.0, 5.0.
+
+
+### Export Behaviors
+
+#### Model Hierarchy and Kind
+
+`MayaUSDExportCommand` currently attempts to make each root prim in the
+exported USD file a valid model, and may author a computed `kind`
+on one or more prims to do so. In the future, we plan to support
+annotating desired `kind` in the Maya scenegraph, and possibly make
+further `kind`/model inference optional. The current behavior is:
+
+* We initially author the kinds on prims based on the `-kind` flag
+  in `MayaUSDExportCommand` (see the `MayaUSDExportCommand` flags above) or the
+  `USD_kind` attribute on individual Maya nodes (see the "Maya
+  Custom Attributes" table below).
+
+* For each root prim in the scene, we *validate* the kind if it
+  has been specified. Otherwise, we *compute* a kind for the root prim:
+  * If the root prim is specified to be an `assembly` (or type derived
+    from `assembly`), then we check to make sure that Maya has not
+    created any gprims (UsdGeomGprim-derived prim-type) below the
+    root prim. If there are any gprims below the root prim, then
+    `MayaUSDExportCommand` will halt with an error.
+  * If the root prim has no kind, then we will compute a value
+    for the kind to ensure that all root prims have a kind.
+    We determine whether a root prim represents a `component`
+    (i.e. leaf model) or `assembly` (i.e. aggregate model of
+    other models, by reference) by determining whether Maya
+    directly created any gprims (UsdGeomGprim-derived prim-type).
+    If Maya has created gprims, model is a `component`, else
+    it is an `assembly`.
+
+* Once we have validated or set the kind on each root prim, we go
+  through each root prim's sub-hierarchy to make sure that it is
+  a valid model:
+  * If model is a `component`, but also has references to other
+    models contained (nested) within it, override the `kind` of
+    the nested references to `subcomponent`, since `component`
+    models cannot contain other models according to USD's
+    model-hierarchy rules.
+  * Else, if model is an `assembly`, ensure that all the intermediate
+    prims between the root and the locations at which other assets
+    are referenced in get `kind=group` so that there is a contiguous
+    hierarchy of models from the root down to the "leaf" model references.
+
+
+#### UV Name Translation: `map1` becomes `st`
+
+Currently, for Mesh export (and similarly for NurbsPatch, also),
```

Comment:
@JGamache-autodesk can you please review this section.

Reply:
Oups, sorry about that. This section should have been updated with the UV set fixes of #876.

The current behavior will preserve uvSet names.
To get back the old renaming behavior requires setting environment variables:
on export, renaming requires MAYAUSD_EXPORT_MAP1_AS_PRIMARY_UV_SET to be set.
on import, renaming depends on MAYAUSD_IMPORT_PRIMARY_UV_SET_AS_MAP1.